internal/driver/glfw/loop.go

```diff
@@ -36,9 +36,9 @@ func runOnMain(f func()) {
 
 // force a function f to run on the main thread and specify if we should wait for it to return
 func runOnMainWithWait(f func(), wait bool) {
-	// If we are on main just execute - otherwise add it to the main queue and wait.
-	// The "running" variable is normally false when we are on the main thread.
-	if !running.Load() {
+	// If we are on main during app run just execute - otherwise add it to the main queue and wait.
```

Comment:
I think this should be "If we are on main _before_ app run..."

Reply:
Thanks @dweymouth , fixed